cachi2/core/package_managers/cargo/main.py

```diff
@@ -34,7 +36,17 @@ def _resolve_cargo_package(package_dir: RootedPath, output_dir: RootedPath) -> l
     cmd = ["cargo", "vendor", "--locked", str(vendor_dir)]
     log.info("Fetching cargo dependencies at %s", package_dir)
     run_cmd(cmd=cmd, params={"cwd": package_dir})
-    return []
+
+    downloaded_deps = os.listdir(vendor_dir)
+    all_configs = [vendor_dir.path / x / "Cargo.toml" for x in downloaded_deps]
+    all_tomls = [TOMLFile(x).read() for x in all_configs]
```

Comment:
This can potentially be a list of fairly sized objects, can we make it lazily evaluated with `map` instead? Especially if you use `iterdir` from above then the 3 lines above can be condensed to a single `map` call.